modules/intensity_transform/src/intensity_transform.cpp

```diff
@@ -0,0 +1,76 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+
+using namespace cv;
+using namespace std;
+
+namespace cv {
+namespace intensity_transform {
+
+void logTransform(const Mat input, Mat& output)
+{
+    double maxVal;
+    minMaxLoc(input, NULL, &maxVal, NULL, NULL);
+    const double c = 255 / log(1 + maxVal);
+
+    if (input.channels() == 1)
+    {
+        output = input + 1;
+    }
+    else
+    {
+        output = input + Scalar(1, 1, 1);
+    }
+
+    output.convertTo(output, CV_64F);
+    cv::log(output, output);
+    output *= c;
+    output.convertTo(output, CV_8UC3);
+}
+
+void gammaCorrection(const Mat input, Mat& output, const float gamma)
+{
+    vector<int> table(256);
+    for (int i = 0; i < 256; i++)
+    {
+        table[i] = (int)(pow((i / 255.0), gamma) * 255.0);
+    }
+
+    LUT(input, table, output);
+    output.convertTo(output, CV_8UC3);
+}
+
+void autoscaling(const Mat input, Mat& output)
+{
+    double minVal, maxVal;
+    minMaxLoc(input, &minVal, &maxVal, NULL, NULL);
+    output = 255 * (input - minVal) / (maxVal - minVal);
+}
+
+void contrastStretching(const Mat input, Mat& output, const int r1, const int s1, const int r2, const int s2)
+{
+    vector<int> table(256);
```

Comment:
`std::array` ?

Reply:
Thank you for your comment. Since cv::LUT method takes in cv::_InputArray as the argument, std::array didn't work here.

Reply:
It should work, related code is [here](https://github.com/opencv/opencv/blame/4.0.0/modules/core/include/opencv2/core/mat.hpp#L207).

BTW, Use `std::array<uchar, 256> table;` and remove final `.convertTo()`.

Reply:
Thanks for the document! But, when I use `std::array<uchar, 256> table;`, I get the following error:
`candidate function not viable: no known conversion from
      'std::array<uchar, 256>' (aka 'array<unsigned char, 256>') to
      'const cv::_InputArray' for 2nd argument
CV_EXPORTS_W void LUT(InputArray src, InputArray lut, OutputArray dst);`
Any idea why?

Reply:
Which compiler is used?
Which version of code from main "opencv" repository is used?

Could you try `LUT(..., InputArray(table), ...)`?

Reply:
This commit works well (pass CI checks, see #2366): https://github.com/opencv/opencv_contrib/commit/b05b32850c924e3465acfcb2d637dd2765928720

Reply:
Great, thank you! Made the change.